src/app/navigation/vertical-navigation.component.ts

```diff
@@ -0,0 +1,665 @@
+    if (collapsedItem) {
+      collapsedItem.secondaryCollapsed = setCollapsed;
+    }
+    if (setCollapsed) {
+      this.collapsedSecondaryNav = true;
+
+      this.addClass('collapsed-secondary-nav-pf');
+    } else {
+      // Remove any collapsed secondary menus
+      if (this.items) {
+        this.items.forEach( (item)  => {
+          item.secondaryCollapsed = false;
+        });
+      }
+      this.collapsedSecondaryNav = false;
+
+      this.removeClass('collapsed-secondary-nav-pf');
+    }
+  }
+
+  private updateTertiaryCollapsedState(setCollapsed: boolean, collapsedItem?: NavigationItemConfig): void {
+    if (collapsedItem) {
+      collapsedItem.tertiaryCollapsed = setCollapsed;
+    }
+    if (setCollapsed) {
+      this.collapsedTertiaryNav = true;
+
+      this.addClass('collapsed-tertiary-nav-pf');
+      this.updateSecondaryCollapsedState(false);
+    } else {
+      // Remove any collapsed secondary menus
+      if (this.items) {
+        this.items.forEach( (item) => {
+          if (item.children && item.children.length > 0) {
+            item.children.forEach( (secondaryItem) => {
+              secondaryItem.tertiaryCollapsed = false;
+            });
+          }
+        });
+      }
+      this.collapsedTertiaryNav = false;
+
+      this.removeClass('collapsed-tertiary-nav-pf');
+    }
+  }
+
+  /**
+   * Clear all active items
+   */
+  public clearActiveItems(): void{
+    this.items.forEach( (item) => {
+      item.isActive = false;
+      if (item.children) {
+        item.children.forEach( (secondary) => {
+          secondary.isActive = false;
+          if (secondary.children) {
+            secondary.children.forEach( (tertiary) => {
+              tertiary.isActive = false;
+            });
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Sets the active items in the vertical navigation
+   */
+  public setActiveItems(): void {
+    let updatedRoute = this.router.url;
+    // //Setting active state on load
+    this.items.forEach( (topLevel)  => {
+      if (updatedRoute.indexOf(topLevel.href) > -1) {
+        topLevel.isActive = true;
+      }
+      if (topLevel.children) {
+        topLevel.children.forEach( (secondLevel) => {
+          if (updatedRoute.indexOf(secondLevel.href) > -1) {
+            secondLevel.isActive = true;
+            topLevel.isActive = true;
+          }
+          if (secondLevel.children) {
+            secondLevel.children.forEach( (thirdLevel) => {
+              if (updatedRoute.indexOf(thirdLevel.href) > -1) {
+                thirdLevel.isActive = true;
+                secondLevel.isActive = true;
+                topLevel.isActive = true;
+              }
+            });
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Public resize event called when the window size changes
+   * @param event
+   */
+  public onResize(event: any): void {
```

Comment:
Does this or any of the following methods need to be public? (I typically try to make everything private, then see what AOT complains about.) I'm asking because the TS linter is complaining about the public methods appearing after the private ones.

I also built the code and added it to my Angular CI app. There are a few errors regarding private methods. For example:

ERROR in ng:///Users/dlabrecq/Work/patternfly/test/patternfly-ng-notification/node_modules/patternfly-ng/src/app/navigation/vertical-navigation.component.d.ts.VerticalNavigationComponent.html (2,722): Property 'activeSecondary' is private and only accessible within class 'VerticalNavigationComponent'.